runhouse/resources/functionals/mapper.py

```diff
@@ -0,0 +1,137 @@
+import logging
+
+import ray
+
+from runhouse.resources.functions import function, Function
+
+from runhouse.resources.module import Module
+
+logger = logging.getLogger(__name__)
+
+
+class Mapper(Module):
+    def __init__(self, module, method, num_replicas=-1, replicas=None, **kwargs):
+        super().__init__(**kwargs)
+        self.module = module
+        self.method = method
+        self.num_replicas = num_replicas
+        self._auto_replicas = []
+        self._user_replicas = replicas or []
+        self._last_called = 0
+        if self.num_replicas > len(self.replicas) and self.num_replicas > 0:
+            self._add_auto_replicas(self.num_replicas - len(self.replicas))
```

Comment:
i'd say get rid of this helper and do it here, more clear

Reply:
Oh that reminds me, I meant to do it in add_replicas too.